src/components/purpose-card.js

```diff
@@ -75,10 +77,76 @@ class PurposeCard extends React.Component {
     })
   }
 
+    handleEditorChange = (content) => {
+    const { activeGroup, field } = this.props
+    const { data } = activeGroup
+
+    this.setState({[field]: content.level.content})
+
+  }
+
+  rtffield = () => {
+    const{ error, message, success } = this.state
+    const { activeGroup, heading, field } = this.props
+    const { data, inEditMode, hasEditRights } = activeGroup
+    return(
+      <Editor
+        apikey="fb3pb0ana4mvi60jwhefs3g2u3501d9s915efud2rh6ax2ek"
+        init={{
+          menubar: false,
+          plugins: [
+            'advlist autolink lists link image charmap print preview anchor',
+            'searchreplace visualblocks code fullscreen',
+            'insertdatetime media table paste code help wordcount'
+            ],
+        }}
+        value={this.state[field]}
+        textareaName={field}
+        onChange={this.handleEditorChange}
+      />
+    )
+  }
+
+
+  textfield = () =>{
+    const { error, message, success } = this.state
+    const { activeGroup, heading, field } = this.props
+    const { data, inEditMode, hasEditRights } = activeGroup
+    return( 
+	    <TextArea
+              autoHeight
+              name={field}
+              value={this.state[field]}
+              onChange={this.handleChange}
+            />)
+  }
+
+
   render () {
     const { error, message, success } = this.state
     const { activeGroup, heading, field } = this.props
     const { data, inEditMode, hasEditRights } = activeGroup
+            
+    let toedit;
+    let display;
+    if({ field }.field =="shortDescription")
```

Comment:
Instead of hardcoding like this, you can pass a prop called renderer=RTField to `<PurposeCard heading='Short description' field='shortDescription' />` in `group-about.js`.

Follow something like this:
1. Make a directory inside components called `fields`.
2. Create three files - rt-field, text-field and index.js
3. `index.js` will export all such fields as a map.
4. Import it in about group to pass it as a prop.